### PR TITLE
fix(python): use download_as_text instead of as_string

### PR DESCRIPTION
**Pull request by glasnt**
`download_as_string` is deprecated, and is a wrapper for `download_as_bytes`

However, `download_as_text` does the decoding for us, so we should be able to use that .

### CODE REVIEW

File: infra/main.tf [hunk 1]
```diff
@@ -17,7 +17,7 @@
 locals {
   unique = "${var.deployment_name}-${random_id.default.hex}"
 
-  application_image = "us-docker.pkg.dev/hsa-public/containers/cloud-client-api/${lower(var.language)}:${var.image_version}"
+  application_image = "us-docker.pkg.dev/hsa-public/containers/cloud-client-api/${lower(var.language)}:${var.image_version}-pr105"
```
Comment (glasnt) [inline infra/main.tf:20]: Revert this before merging; but also: Update the tutorial text to match this method name change. 

```suggestion
  application_image = "us-docker.pkg.dev/hsa-public/containers/cloud-client-api/${lower(var.language)}:${var.image_version}"
```

**Review by NimJay — APPROVED** [review]
Checks are passing. Approved.

**Pull request merged**